src/datasets/data_files.py

```diff
@@ -748,6 +749,18 @@ def from_local_or_remote(
         origin_metadata = _get_origin_metadata_locally_or_by_urls(data_files, use_auth_token=use_auth_token)
         return cls(data_files, origin_metadata)
 
+    def filter_extensions(self, extensions: List[str]) -> "DataFilesList":
+        pattern = "|".join("\\" + ext for ext in extensions)
+        pattern = re.compile(f".*({pattern})(\\..+)?$")
+        return DataFilesList(
+            [
+                data_file
+                for data_file in self
+                if pattern.match(data_file.name if isinstance(data_file, Path) else data_file)
+            ],
+            origin_metadata=self.origin_metadata,
+        )
```

Comment:
Here we should also drop the origin metadata of the removed data files, no?

Reply:
`origin_metadata` is the list of origin per **pattern**, not per data file. We don't know which pattern generated which data file, and a pattern may have generated multiple data files. So I don't think we can easily drop origin metadata during filtering.

Note that for packaged builders patterns often look like "**" or "train/**" so there's only a few origin_metadata. Adding or removing new unsupported files in the dataset repo won't change the origin metadata, and they'll be filtered out.